### DISCUSSION TITLE
How do I stop the junk?

### DISCUSSION THREAD

**Discussion by nelgin** [dosemu2/dosemu2#2053] [Q&A]
I am trying to get dosemu2 working on my bulletin board again. 

I would like to find a way to subdue all these startup messages. I've tried using -q with dosemu.bin but it doesn't seem to work.

[https://pasteboard.co/FizwWw7q63G5.png](url)

I just want to load dosemu without any of the pretext and then launch the game.

This startup command has been given to me so may need tweaking.

/usr/bin/env  HOME=/sbbs/ctrl/ QUIET=1 DOSDRIVE_D=/sbbs/node1/ DOSDRIVE_E=/sbbs/xtrn DOSDRIVE_F=/sbbs/ctrl/ DOSDRIVE_G=/sbbs/data/ DOSDRIVE_H=/sbbs/exec/ NODEDIR=/sbbs/node1/ /usr/bin/dosemu.bin SHELL_LOADHIGH_DEFAULT=0 -d /sbbs/node1/ -d /sbbs/xtrn -d /sbbs/ctrl/ -d /sbbs/data/ -d /sbbs/exec/ -d /sbbs/xtrn -I"video { none }" -I"serial { virtual com 1 }" -f/sbbs/ctrl/dosemu.conf -ED:external.bat -o/sbbs/node1/dosemu_boot.log

open to suggestions.

**Comment by stsp** [comment]
You can use `dosemu -quiet` to suppress
startup messages. However as you decided
to invoke dosemu.bin directly, you are on
your own (as usual).

**Comment by nelgin** [comment]
I replaced $DOSEMUBIN with /usr/bin/dosemu -quiet and that seemed to work, except only half of my games seem to work. Those that did work seemed to display slowly sometimes. maybe that's a function of the BBS, I'll have to check.

**Reply by stsp** [reply]
You can try `speed 0` to give some
more cpu power to the DOS process.

**Comment by nelgin** [comment]
speed 0 goes after "virtual com 1" ?

**Reply by stsp** [reply]
`speed 0` is a command you need to
run in DOS. Alternatively you can set
`$_hogthreshold = (0)` which is the same.

**Comment by nelgin** [comment]
That seemed to work. Thank you.

**Comment by stsp** [comment]
In that case you need to try more
cpu-saving values, like 2 or 5.

**Comment by nelgin** [comment]
It was set to 1 already.

**Comment by stsp** [comment]
But if 1 was too slow then you need
something else. You need to increase
the value, but there is a discontinuity
and 0 means maximum :(

**Comment by nelgin** [comment]
Both 2, 3 and 5 all seem to work fine and display as quick as 0.

**Comment by stsp** [comment]
It would be good if, before your next
return, you'll change this:
```
/usr/bin/env HOME=/sbbs/ctrl/ QUIET=1 DOSDRIVE_D=/sbbs/node1/ DOSDRIVE_E=/sbbs/xtrn DOSDRIVE_F=/sbbs/ctrl/ DOSDRIVE_G=/sbbs/data/ DOSDRIVE_H=/sbbs/exec/ NODEDIR=/sbbs/node1/ /usr/bin/dosemu.bin SHELL_LOADHIGH_DEFAULT=0 -d /sbbs/node1/ -d /sbbs/xtrn -d /sbbs/ctrl/ -d /sbbs/data/ -d /sbbs/exec/ -d /sbbs/xtrn -I"video { none }" -I"serial { virtual com 1 }" -f/sbbs/ctrl/dosemu.conf -ED:external.bat -o/sbbs/node1/dosemu_boot.log
```
into something more comprehensive.
For example you can add extra drives
with `-d` option, rather than to set up
the env var and then presumably using
`lredir` on that.

**Comment by nelgin** [comment]
Actually, no, we're not using lredir any more so those environment variables are obsolete. I've removed them and now have:

/usr/bin/env  HOME=/sbbs/ctrl/ QUIET=1 NODEDIR=/sbbs/node1/ /usr/bin/dosemu -quiet SHELL_LOADHIGH_DEFAULT=0 -d /sbbs/node1/ -d /sbbs/xtrn -d /sbbs/ctrl/ -d /sbbs/data/ -d /sbbs/exec/ -d /sbbs/xtrn -I"video { none }"  -I"serial { virtual com 1 }" -f/sbbs/ctrl/dosemu.conf -ED:external.bat -o/sbbs/node1/dosemu_boot.log

The good news is that with this configuration previously unworking games Yankee Trader and Trade Wars are now working.

Unfortunately, previously workings games are not. It's late so I'll do some debugging later.

**Reply by stsp** [reply]
I suppose `SHELL_LOADHIGH_DEFAULT=0` is
also now unneeded.
`-I"serial { virtual com 1 }"` should be replaced
with `$_com1 = "virtual"` in a config.
`-I"video { none }"` should be replaced with `-dumb`.
Additionally, all `-d`s can be compacted into a
`$_hostfs_drives =` setting.

**Reply by stsp** [reply]
`QUIET=1` is also unneeded if you have `-quiet`.

**Reply by stsp** [reply]
Overall, the fewer customizations you do,
the easier is to help you finding a problem,
once it appear.
Of course there is the opposite factor, which
is if no one tests the advanced configs, they
are getting quickly broken. But in your case
its definitely better to "stabilize" things with
as few customizations as possible first, and
only then go for something more advanced.

**Comment by nelgin** [comment]
The good news on the problem programs it that it was a missing config from the autoexec.bat or whatever it's called these days. When I copied the new environment I forgot to pop in the path and an environment variable so problem solved.

I can do -quiet, -dumb and $_com1 settings, I can't use $_hostfs_drives since they're dynamically generated by the BBS based on the install location. Not everybody installs the BBS in the same location and I'm trying to create a dropin that "just works" for everyone. I agree, the less configuration parameters the better to keep things easy to understand. I have no idea what dosemu.bin was used in the first place. That goes way back to the dosemu 1.4 days and was done before I even first used the BBS system that integrates it.

With a number of programs, notably Tradewars, not working with dosemu and with dosemu2 now easy to install through ppa, it's worth a little effort to get it working.

Thanks for your support and suggestions.

**Reply by stsp** [reply]
Yes, good point about `-d`.
Maybe you can then write a wrapper
script that will automatically generate
the needed command line and start
dosemu2, taking only bbs root dir as
an input.

Also you may be interested to look here:
https://github.com/dosemu2/dosemu2/discussions/1983#discussioncomment-5560747
That guy succeeded with `comredir`
for bbs software, replacing "virtual"
hack of dosemu1.
Its not like I am going to remove "virtual"
before dosemu3 or alike, but migrating
to dosemu2 techniques is always welcome.
I myself wouldn't introduce "virtual", but
since its already there, it of course can't
be just removed.

**Reply by nelgin** [reply]
I'm not sure why I would want to use comredir. It looks like I have to setup com ports with null modems? Seems like adding more work and another layer of complexity that we don't really need when virtual seems to work perfectly fine.

**Reply by stsp** [reply]
Well, that guy was pointing to many problems
of "virtual"... Although since they are now fixed,
maybe there is indeed no point to migrate for you.
But "virtual" was also a source of all your
problems in the past, when you didn't see
the boot process. Of course after that I made
a boot process visible (which can now be disabled
with `-quiet`), so arguably even that no longer
stands.

**Reply by stsp** [reply]
https://github.com/dosemu2/dosemu2/discussions/1983#discussioncomment-5574180
Note this comment of him.
He wants to simulate disconnect, which
is not possible with "virtual". Although since
he himself moved to "virtual", that remained
unimplemented also for comredir. :)

**Reply by nelgin** [reply]
I wouldn't need to simulate a disconnect there. At this stage, which the config I have, everything seems to be working, even stuff that wasn't working dosemu1 and the earlier attempts I made to use dosemu2. don't touch anything! :)

I've got it down to this, This is expanded from a dosemu.ini template file that is read by the BBS. $NODEDIR, $XTRNDIR, $CTRLDIR etc are all set within the BBS and passed through, parsed and then used to launch dosemu. I think this is a good balance.

/usr/bin/env  HOME=/sbbs/ctrl/ NODEDIR=/sbbs/node1/ /usr/bin/dosemu -quiet -dumb SHELL_LOADHIGH_DEFAULT=0 -d /sbbs/node1/ -d /sbbs/xtrn -d /sbbs/ctrl/ -d /sbbs/data/ -d /sbbs/exec/ -d /sbbs/xtrn -f/sbbs/ctrl/dosemu.conf -ED:external.bat -o/sbbs/node1/dosemu_boot.log

**Reply by stsp** [reply]
So does `SHELL_LOADHIGH_DEFAULT=0` really help?

**Reply by stsp** [reply]
Also not sure what `HOME=/sbbs/ctrl/ NODEDIR=/sbbs/node1/`
are good for. If you put them at the end of the command line,
then they'll be set as a DOS env variables. But in the
beginning of cmd line they are unix env variables, so
I wonder how you use those.

**Reply by stsp** [reply]
> I've got it down to this, This is expanded from a dosemu.ini template file that is read by the BBS. $NODEDIR,
> $XTRNDIR, $CTRLDIR etc are all set within the BBS and passed through, parsed and then used to launch
> dosemu.

Perhaps in that case you'd want
`-d $CTRLDIR` and so on, instead of the
fixed values?

**Comment by nelgin** [comment]
That's what I do have:

cmd=/usr/bin/env $TERM $DOSEMUBIN -quiet -dumb -d $NODEDIR -d $XTRNDIR -d $CTRLDIR -d $DATADIR -d $EXECDIR -d $XTRNDIR -f$DOSEMUCONF -E$EXTBAT -o$NODEDIRdosemu_boot.log $EXTLOG

but those variables are generated within the BBS software based on the sysop's configuration.

I think HOME and CTRL are leftovers from dosemu1. I think they were used to dynamically create one of the config files.
SHELL_LOADHIGH_DEFAULT=0 doesn't appear to be needed any more. I think someone said that it was the default now anyway.

**Reply by stsp** [reply]
Yes, its default unless `-T` is used, and
was not supposed to be user-tweakable
anyway. I just let you know about it for
debugging problems.

**Comment by nelgin** [comment]
Good to know. It's gone now. I think it looks good now. Less wieldy.

**Reply by stsp** [reply]
The reason for all that nit-picking is that
you write a howto. It doesn't matter what
you do, as much as it does to have a proper
strings in a howto. :)

**Comment by nelgin** [comment]
Like I said, a lot of this stuff was done before me and using dosemu1 so much is left over "legacy" stuff. Some stuff is built into the BBS that we can't easily change or it'll break other installations. Eventually it'd be nice to get everyone using dosemu2 from ppa (for those on Ubuntu anyway) to make it easier to support and made door installation more portbale.

**Reply by stsp** [reply]
I hope you don't need to rewrite the
dosemu1 doors howto, and can just
introduce a new one, for those who
already get dosemu2 from PPA.
Given that now you seem to have
the legacy-free setup, I hope such
plan should work.

**Comment by nelgin** [comment]
Yup, but it'll be only relevant to the Synchronet bbs setup but may work with others.

**Reply by stsp** [reply]
Was dosemu1-related howto
more complete than that?

**Comment by nelgin** [comment]
I don't recall to be honest. I followed the synchronet instructions to get it working and it just sort of worked, I think.

**Reply by nelgin** [reply]
I figured out what $HOME was for.

HOME=/sbbs/ctrl/ is where the .dosemu setup is with preconfigured autoexec.bat and config.sys (or whatever they are now) so yes, that does appear to be needed.

**Reply by stsp** [reply]
How is so?
You have `-f/sbbs/ctrl/dosemu.conf`, which is
relevant. HOME= is completely irrelevant.

**Reply by nelgin** [reply]
Using  HOME=/sbbs/ctrl /usr/bin/dosemu -d /sbbs/node1/ -d /sbbs/xtrn -d /sbbs/ctrl/ -d /sbbs/data/ -d /sbbs/exec/ -d /sbbs/xtrn -f/sbbs/ctrl/dosemu-local.conf

I get

![image](https://github.com/dosemu2/dosemu2/assets/6232589/b464713c-1e6b-47c1-bf23-f98809c583cd)

and if I don't...

![image](https://github.com/dosemu2/dosemu2/assets/6232589/bce42c75-5c21-410e-baab-036c4f87640e)

So tell me why is $HOME irrelevant?

**Reply by stsp** [reply]
OK, sorry, HOME is not irrelevant.
Its just irrelevant for half of what you said
in `the .dosemu setup is with preconfigured autoexec.bat and config.sys`.
Namely, the autoexecs/configs are in
`/usr/share/dosemu/commands`, and
as for `.dosemu` - I misread it as `.dosemurc`,
for which -f is relevant.
Anyway, instead of altering HOME, you can
try `--Flocaldir /sbbs/ctrl`, which should be the
dosemu2 equivalent. Let me know if it doesn't
work as expected that way.

**Reply by stsp** [reply]
Another possibility should be
`--Fdrive_c /sbbs/ctrl/.dosemu/drive_c`
as you already specify -f and -o, so
everything except drive_c should already
be moved.

**Reply by stsp** [reply]
And actually I have just renamed `--Flocaldir`
to `--Flocal_dir` to be consistent with
`$_local_dir` option and follow the same style
with `--Fdrive_c`.
This option is very new, hope no one used it
yet. :)

**Comment by nelgin** [comment]
If you just did it then I’ll wait for PPA to update and maybe try it tomorrow.

Thanks,

**Reply by stsp** [reply]
Well you can try the `--Fdrive_c` way
w/o any waiting. `--Flocal_dir` may only need
waiting, but I don't think you need it if
`--Fdrive_c` is enough.
